### ISSUE TITLE
swap buttons in 'remove repository' and 'discard changes' dialog on Windows

### ISSUE THREAD

**Issue by iridinite** [desktop/desktop#2463]
### Description

I would like to suggest swapping the Cancel and Remove buttons in the confirmation dialog for removing a repo from Desktop. The reason is that in other dialogs, such as branch creation, cloning and creating a new repo, the button to proceed/confirm is always on the left side and colored blue.

![image](https://user-images.githubusercontent.com/20139189/29289607-64276936-813d-11e7-87a2-9ca8402fd300.png)

For the sake of consistency, I think this should go for the repo removal confirmation as well. It's quite easy for a user who might not exactly be paying attention, to hit the wrong button, because they were expecting Cancel to be on the other side.

I guess it might be nitpicky, but this really struck me as an odd design decision. :smile: If this was done intentionally to draw the user's attention to the Cancel button (as removing something might be destructive), might I suggest instead coloring the Remove button red, like the 'danger zone' buttons on a repo's controls on the GH site?

Other dialog boxes, to illustrate my point: (Note that the "yes, continue"-like option is colored and is on the left.)

![image](https://user-images.githubusercontent.com/20139189/29289751-f29ffce6-813d-11e7-8e4e-c046751b1e0f.png)
![image](https://user-images.githubusercontent.com/20139189/29289780-0e4d4d22-813e-11e7-8bf5-b31d18825072.png)

### Version

**GitHub Desktop version:** 0.7.2

**OS version:** Windows 10 Home

**Comment by j-f1** [comment]
It looks like the “Discard Changes” dialog has the “Cancel” button in the same position:

<img width="428" alt="screen shot 2017-08-14 at 17 49 39" src="https://user-images.githubusercontent.com/25517624/29293406-07bd3eec-8119-11e7-94d2-3e32e1adfafc.png">

**Comment by shiftkey** [comment]
Walking through the first example...

The button group for `confirm-remove-repository` defaults to Cancel for the submit behaviour:

https://github.com/desktop/desktop/blob/4791d54f705830b064b47276e1a4c7dd6d632f31/app/src/ui/remove-repository/confirm-remove-repository.tsx#L51-L54

We're not allowed to put the submit button last, because we have [this rule](https://github.com/desktop/desktop/blob/4791d54f705830b064b47276e1a4c7dd6d632f31/tslint-rules/buttonGroupOrderRule.ts) that expects the submit button to go first, for consistency when implmenting dialogs.  

When rendering the button group it's this code that determines when to flip the buttons around on Windows:

https://github.com/desktop/desktop/blob/4791d54f705830b064b47276e1a4c7dd6d632f31/app/src/ui/lib/button-group.tsx#L50-L55

So we're enforcing the submit action is rendered first, which on macOS does make sense - but maybe it's not as logixal on Windows, when you compare it to the create and clone dialogs.

I need to refresh my memory on what Windows expects for these sorts of "destructive" actions.

**Comment by shiftkey** [comment]
An assortment of example Windows dialogs:

<img width="376" src="https://user-images.githubusercontent.com/359239/29300716-9a2b4dfe-81ba-11e7-9328-64244aa83556.png">

<img width="380" src="https://user-images.githubusercontent.com/359239/29300719-9b43890e-81ba-11e7-99e0-af301d31134c.png">

<img width="410" src="https://user-images.githubusercontent.com/359239/29300718-9a5b21be-81ba-11e7-99e2-8b7f6e341561.png">

<img width="346" src="https://user-images.githubusercontent.com/359239/29300714-9a05516c-81ba-11e7-88de-975069956844.png">

(this last one is Sublime)

What's interesting here is the "Confirm" action is focused, and pressing "enter" to submit would perform the action. This is inverse to the macOS convention and how these dialogs work currently. Need to ponder a way out of this that works with our conventions.

**Comment by shiftkey** [comment]
Putting this in the design queue to get their feedback on changing these forms on Windows.

**Comment by j-f1** [comment]
> What's interesting here is the "Confirm" action is focused, and pressing "enter" to submit would perform the action. This is inverse to the macOS convention

The native default buttons on macOS get activated by <kbd>return</kbd>, too.

**Comment by iridinite** [comment]
If the intention is to warn users about a potentially dangerous action, how about making the confirmation option red?

![image](https://user-images.githubusercontent.com/20139189/29314116-d62aefba-81bc-11e7-97c1-9b9e1ba09dbe.png)

Just trying to be helpful here ^^

**Comment by joshaber** [comment]
I think the question here is specific to Windows.

To provide some context on our behavior on macOS: when an action is destructive (e.g., discard changes), the convention is that the default button should be `Cancel`. Otherwise the default button should be the action button.

So the question is, what's the Windows convention? In @shiftkey's examples, those all seem to be non-destructive actions, so I'm not sure they give us the answer.

**Comment by shiftkey** [comment]
@joshaber those screenshots are from dialogs where you are about to lose unsaved changes to a file, and they are favouring the safe option of saving the file, rather than the destructive result of discarding...

**Comment by joshaber** [comment]
Ahhh, I missed that 👍

**Comment by bjcull** [comment]
The confirmation button should always be the coloured one (and in my opinion always on the left, for windows users at least). I almost accidentally deleted a branch precisely because I snapped to the right hand side for a cancel.

![image](https://user-images.githubusercontent.com/184263/30239160-bf1333d4-9556-11e7-82ac-bb27a643bd25.png)

This type of action is different to the examples given above, as it's the primary action and not a side effect of something else (like closing a file and losing changes). An example of a direct action like this is permanently deleting a file in windows. 

![image](https://user-images.githubusercontent.com/184263/30239228-29e2c976-9558-11e7-8fe9-717304652447.png)

The buttons are in the correct order and the safe action is outlined by default. My only change would be to colour the primary action red, as it's destructive.

**Comment by joshaber** [comment]
It sounds like everyone's agreed: the confirmation button should always be on the left, where currently it's on the right for destructive actions.

**Comment by timofonic** [comment]
@bjcull @joshaber Yes, I think it's should be on the left too. Any changes in the code? Why is this and the PR still open and not merged? :P

--- It seems the usual convention in Japanese and Chinese is different-----
Anyone knows Chinese/Japanese or is Chinese/Japanese in the team? I used some Chinese software (QQ) despite having no idea of the language. It seems Chinese software "ok/cancel/whatever" buttons are ordered differently than in other languages. I think to remember the same happens to Japanese too, but I don't remember it clearly.

Can this be confirmed? Would this be useful when localizating the software to those languages?

**Comment by shiftkey** [comment]
We took a shot at this in https://github.com/desktop/desktop/pull/3347 but unfortunately encountered some issues with how the Dialogs currently work - this is summarized in https://github.com/desktop/desktop/pull/3347#discussion_r153976000.

**Comment by shiftkey** [comment]
Dropping `help-wanted` on here while we figure out how to proceed.

**Comment by billygriffin** [comment]
@niik and I are working on this, and because it became really hard to follow all the potential cases for both macOS and Windows, we wanted to share this diagram to help guide our path. We'll also include this in docs about how these should work:

![button conventions](https://user-images.githubusercontent.com/5091167/68219886-f794ff80-ffa3-11e9-9f25-40a9bb2e9a71.png)

**Comment by bjcull** [comment]
I think having the cancel button in blue is still weird, but putting it in the right place is a big improvement, thank you!

**Comment by billygriffin** [comment]
Thanks @bjcull! The blue is our way of depicting which button gets initial focus, and therefore would be selected when hitting `Enter` or `Spacebar`. Totally understand that might feel odd to some folks, but glad it's in the right direction. 😄
